src/04-modality-specific-files/01-magnetic-resonance-imaging-data.md

```diff
@@ -104,6 +107,24 @@ Useful for multimodal co-registration with MEG, (S)EEG, TMS, etc.
 
+### Echo-Planar Imaging & B0 field-mapping
+
+Echo-Planar Imaging (EPI) schemes typically used in the acquisition of
+diffusion and functional MRI may also be *intended for* estimating the
+*B=0* field inside the scanner (in other words, *estimating the fieldmap*,)
+without the acquisition of additional MRI schemes such as gradient-recalled
+echo (GRE) sequences that are stored under the `fmap/` folder of the
+BIDS structure.
+
+The modality labels `dwi` (under `dwi/`), `bold`, and `sbref` (under `func/`)
+and any modality under `fmap/` are allowed to encode the MR protocol intent for
+fieldmap estimation using the following metadata:
+
+| **Key name**                | **Requirement level** | **Data type**                         | **Description**                                                                                                                                                                                                                                                                                                                                                                                                                                     |
+| --------------------------- | --------------------- | ------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| B0FieldIdentifier           | OPTIONAL              | [string][]                            | The presence of this key states that this particular 3D or 4D image may be used for fieldmap estimation purposes. The `B0FieldIdentifier` MUST be a unique string within one participant's tree, shared only by the images meant to be used as inputs for the estimation of a particular instance of the *B0 field* estimation.                                                                                                                     |
+| B0FieldSource               | OPTIONAL              | [string][] or [array][] of [string][] | At least one existing `B0FieldIdentifier` defined by other images in the participant's tree. This field states the *B0 field* estimation designated by the `B0FieldIdentifier` that may be used to correct the dataset for distortions caused by B0 inhomogeneities. `B0FieldSource` and `B0FieldIdentifier` are mutually exclusive. When a list of identifiers is given, the estimated fieldmap may be averaged across the different sources.      |
```

Comment:
Why use unique identifiers instead of paths to filenames, as in `IntendedFor`? I originally assumed that the identifier also indicated the type of B0 field map, but that doesn't seem to be the case. Does an identifier add anything over the filename?

Reply:
The key to these identifiers is that they allow the definition of higher-level data constructs.

> Does an identifier add anything over the filename?

See https://github.com/bids-standard/bids-specification/pull/622/files#diff-d668661006216379ab2a5c5ff7061b8aR384 

You could theoretically replace the two identifiers with lists-of-lists:
```
{
  "B0FieldSource": ["phasediff_fmap0", "pepolar_fmap0"]
}
```

But I think that is not:
- Readable
- Easy to encode
- Granular enough: the ID "phasediff_fmap0" defines much better how the estimation of the fieldmap must be done given a subset of the data having that `B0FieldIdentifier` (even more so for the pepolar fieldmap).

Reply:
Easy to encode as in separating concepts:
- What data should I use to estimate a particular fieldmap? and
- what of the estimated fieldmaps available in this dataset can/should be used to correct this particular image?

the ID creates more clear interface between producers and consumers.

Reply:
Another plus: IDs/tags are more reliable against missing data.

Reply:
That makes sense. I can see the utility of identifiers in that case, although I guess one could still use a list of lists to point to filenames. Thank you for the clarification.

Reply:
Two things I think are worth considering:

1. `B0FieldIdentifier` could be replaced by a column in the scans.tsv file (e.g., `phasediff_fmap0`) with a 0 for scans that aren't part of the group and a 1 for scans that are part of the group. I don't know if anyone else likes this idea, but it allows data providers to define arbitrary sets of files for their own purposes.
2. I don't think `B0FieldIdentifier` and `B0FieldSource` should be mutually exclusive. I'm thinking specifically of HCP style acquisitions, where you get separate runs with opposite encoding directions, so you can apply PEPOLAR-style unwarping to them. In that case the functional scan is part of its own B0 field source (along with the other run with the opposite direction) and it needs to have an identifier to that effect.